model/test_results.go

```diff
@@ -295,42 +321,115 @@ func (t *TestResults) Download(ctx context.Context) ([]TestResult, error) {
 		t.ID = t.Info.ID()
 	}
 
-	var results testResultsDoc
-	catcher := grip.NewBasicCatcher()
-	bucket, err := t.GetBucket(ctx)
-	if err != nil {
-		return nil, err
-	}
-
 	switch t.Artifact.Version {
 	case 0:
+		catcher := grip.NewBasicCatcher()
+		bucket, err := t.GetBucket(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		var results []TestResult
 		iter := NewTestResultsIterator(bucket)
 		for iter.Next(ctx) {
-			results.Results = append(results.Results, iter.Item())
+			results = append(results, iter.Item())
 		}
-
 		catcher.Wrap(iter.Err(), "iterating test results")
 		catcher.Wrap(iter.Close(), "closing test results iterator")
+
+		return results, catcher.Resolve()
 	case 1:
-		r, err := bucket.Get(ctx, testResultsCollection)
+		bsonResults, err := t.downloadBSON(ctx)
 		if err != nil {
-			catcher.Wrap(err, "getting test results")
-			break
+			return nil, err
 		}
-
-		data, err := ioutil.ReadAll(r)
-		catcher.Add(r.Close())
-		if err != nil {
-			catcher.Wrap(err, "reading test results")
-			break
+		parquetResults, err := t.downloadParquet(ctx)
+		if err != nil && !pail.IsKeyNotFoundError(err) {
```

Comment:
I'm wondering about the backward compatibility of this change. Won't this error when downloading any previous test results since they're not stored in Presto?

Reply:
Nope because we ignore the `IsKeyNotFoundError`, so 404s are ignored. Anything else (like auth or network) we would want to be error on.

Reply:
Ah yeah, missed the negation on `IsKeyNotFound`.